data/pc/1.6.4/protocol.json

```diff
@@ -0,0 +1,2286 @@
+            "type": "position_isi"
+          },
+          {
+            "name": "action",
+            "type": "i8"
+          },
+          {
+            "name": "nbtData",
+            "type": "compressedNbt"
+          }
+        ]
+      ],
+      "packet_tile_editor_open": [
+        "container",
+        [
+          {
+            "name": "entityId",
+            "type": "i8"
+          },
+          {
+            "name": "position",
+            "type": "position_iii"
+          }
+        ]
+      ],
+      "packet_increment_statistic": [
+        "container",
+        [
+          {
+            "name": "statisticId",
+            "type": "i32"
+          },
+          {
+            "name": "amount",
+            "type": "i32"
+          }
+        ]
+      ],
+      "packet_player_info": [
+        "container",
+        [
+          {
+            "name": "playerName",
+            "type": "string"
+          },
+          {
+            "name": "online",
+            "type": "bool"
+          },
+          {
+            "name": "ping",
+            "type": "i16"
+          }
+        ]
+      ],
+      "packet_abilities": [
+        "container",
+        [
+          {
+            "name": "flags",
+            "type": "i8"
+          },
+          {
+            "name": "flyingSpeed",
+            "type": "f32"
+          },
+          {
+            "name": "walkingSpeed",
+            "type": "f32"
+          }
+        ]
+      ],
+      "packet_tab_complete": [
+        "container",
+        [
+          {
+            "name": "text",
+            "type": "string"
+          }
+        ]
+      ],
+      "packet_settings": [
+        "container",
+        [
+          {
+            "name": "locale",
+            "type": "string"
+          },
+          {
+            "name": "viewDistance",
+            "type": "i8"
+          },
+          {
+            "name": "chatFlags",
+            "type": "i8"
+          },
+          {
+            "name": "difficulty",
+            "type": "i8"
+          },
+          {
+            "name": "showCape",
+            "type": "bool"
+          }
+        ]
+      ],
+      "packet_client_command": [
+        "container",
+        [
+          {
+            "name": "payload",
+            "type": "i8"
+          }
+        ]
+      ],
+      "packet_scoreboard_objective": [
+        "container",
+        [
+          {
+            "name": "name",
+            "type": "string"
+          },
+          {
+            "name": "displayText",
+            "type": "string"
+          },
+          {
+            "name": "action",
+            "type": "i8"
+          }
+        ]
+      ],
+      "packet_scoreboard_score": [
+        "container",
+        [
+          {
+            "name": "itemName",
+            "type": "string"
+          },
+          {
+            "name": "action",
+            "type": "i8"
+          },
+          {
+            "name": "scoreName",
```

Comment:
scoreName and value are are not read or written if action = 1 

See 1.6.4 source or 
https://wiki.vg/index.php?title=Protocol&oldid=4899#Update_Score_.280xCF.29 for more details.

Reply:
How would the specification for this be?

Reply:
It would look something like this https://github.com/PrismarineJS/minecraft-data/tree/master/data/pc/1.7/protocol.json#L2042-L2068